phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -488,7 +488,62 @@ public void rollback(PhoenixConnection connection) throws SQLException {
         numRows = 0;
     }
     
+    private Set<Integer> getUncommittedSattementIndexes() {
+    	Set<Integer> result = newHashSet();
+    	for (Map<ImmutableBytesPtr, RowMutationState> rowMutations : mutations.values()) {
+    		for (RowMutationState rowMutationState : rowMutations.values()) {
+    			result.addAll(rowMutationState.getOrderOfStatementsInConnection());
+    		}
+    	}
+    	return result;
+    }
+    
     @Override
     public void close() throws SQLException {
     }
+    
+    public static class RowMutationState {
+        private Map<PColumn,byte[]> columnValues;
+        private Set<Integer> orderOfStatementsInConnection;
```

Comment:
Rename for consistency with CommitException to statementIndexes here and getter method. I still think a Set is overkill. Can't it just be an int[] of size 1? Then the join can resize it (in the somewhat unlikely event of a collision).